hamocc/mo_aufr_bgc.F90

```diff
@@ -400,13 +400,15 @@ subroutine aufr_bgc(kpie,kpje,kpke,ntr,ntrbgc,itrbgc,trc,kplyear,kplmon,kplday,o
     call read_netcdf_var(ncid,'dms',locetra(1,1,1,idms),2*kpke,0,iotype)
     call read_netcdf_var(ncid,'fdust',locetra(1,1,1,ifdust),2*kpke,0,iotype)
     call read_netcdf_var(ncid,'iron',locetra(1,1,1,iiron),2*kpke,0,iotype)
-    call read_netcdf_var(ncid,'prefo2',locetra(1,1,1,iprefo2),2*kpke,0,iotype)
-    call read_netcdf_var(ncid,'prefpo4',locetra(1,1,1,iprefpo4),2*kpke,0,iotype)
-    call read_netcdf_var(ncid,'prefalk',locetra(1,1,1,iprefalk),2*kpke,0,iotype)
-    call read_netcdf_var(ncid,'prefdic',locetra(1,1,1,iprefdic),2*kpke,0,iotype)
     call read_netcdf_var(ncid,'dicsat',locetra(1,1,1,idicsat),2*kpke,0,iotype)
-    if(lread_prefsi) then
-      call read_netcdf_var(ncid,'prefsilica',locetra(1,1,1,iprefsilica),2*kpke,0,iotype)
+    if (use_pref_tracers) then
+      call read_netcdf_var(ncid,'prefo2',locetra(1,1,1,iprefo2),2*kpke,0,iotype)
+      call read_netcdf_var(ncid,'prefpo4',locetra(1,1,1,iprefpo4),2*kpke,0,iotype)
+      call read_netcdf_var(ncid,'prefalk',locetra(1,1,1,iprefalk),2*kpke,0,iotype)
+      call read_netcdf_var(ncid,'prefdic',locetra(1,1,1,iprefdic),2*kpke,0,iotype)
+      if(lread_prefsi) then
+        call read_netcdf_var(ncid,'prefsilica',locetra(1,1,1,iprefsilica),2*kpke,0,iotype)
+      endif
```

Comment:
Is there a reason why the preformed silicate is treated differently from the other preformed tracers? Does it make sense to have the capability to restart with a preformed tracer when it is not in the restart file only for silicate but not for the others? I do not see so many use cases for this so maybe we can just omit the `lread_prefsi` related code?

Reply:
Hi Jörg, since preformed silicate wasn't in the restart files before, start-up from restart files was an issue, hence this little quirky treatment that was put in to enable running the model from a restart file without it. Once we settle with #356 or have enough spun-up restart files with preformed silicate, this won't be an issue any longer and it can be removed. Hence, I would keep this treatment for now.